extension/observer/ecsobserver/config.go

```diff
@@ -52,6 +53,31 @@ type Config struct {
 	DockerLabels []DockerLabelConfig `mapstructure:"docker_labels" yaml:"docker_labels"`
 }
 
+// Validate overrides the embedded noop validation.
+func (c *Config) Validate() error {
```

Comment:
Make sure any new method added that are only called in this package are private, I think `Validate` is at least. Should do a scrub of existing methods / structs soon too but not for this PR, we should take care of new ones though.

Reply:
This one has to be `Validate` so it will get called when loading config automatically  https://github.com/open-telemetry/opentelemetry-collector/blob/0bbb1b7f3d886c4ba7a03b36ce4b57c538a727c5/config/extension.go#L17-L24

Reply:
I made all the internal validation logic private, so the sub config for matchers are exported but all the methods `validate` and `newMatcher` are private now.